aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -638,25 +724,30 @@ Tensor _embedding_bag_backward(const Tensor &grad, const Tensor &indices,
   if (sparse) {
     return at::_embedding_bag_sparse_backward(
         grad, indices, offsets, offset2bag_, bag_size_, num_weights,
-        scale_grad_by_freq, mode, per_sample_weights);
+        scale_grad_by_freq, mode, per_sample_weights, padding_idx);
   } else {
     return at::_embedding_bag_dense_backward(
         grad, indices, offsets, offset2bag_, bag_size_, max_indices_, num_weights,
-        scale_grad_by_freq, mode, per_sample_weights);
+        scale_grad_by_freq, mode, per_sample_weights, padding_idx);
   }
 }
 
 static Tensor _embedding_bag_dense_backward_cpu_max(
     const Tensor& grad,
     const Tensor& bag_size,
     const Tensor& max_indices,
-    int64_t num_weights) {
+    int64_t num_weights,
+    int64_t padding_idx) {
   AT_ASSERT(max_indices.defined());
   auto index_grad_weight =
       at::zeros({num_weights, grad.size(1)}, grad.options());
   auto nonempty_max_indices = max_indices.index_select(0, bag_size.nonzero().view(-1));
   auto nonempty_grad = grad.index_select(0, bag_size.nonzero().view(-1));
 
+  if (padding_idx >= 0) {
+    nonempty_grad.masked_fill_(nonempty_max_indices.eq(padding_idx), 0);
```

Comment:
I think creating a mask tensor with `.eq()` is probably not ideal. Maybe I can find a more performant way to do this.

Reply:
One option is to use `TensorIterator` to write directly to `nonempty_grad`, avoiding the intermediate mask tensor. Something like:
```python
auto iter = TensorIteratorConfig()
  .add_output(nonempty_grad)
  .add_input(nonempty_max_indices)
  .check_all_same_dtype(false)
  .build();
```

Then either use `iter.for_each()` or dispatched CPU / CUDA kernels with a lambda that captures `padding_idx` and checks it against the index.

Reply:
Welp, it turns out that the best solution is actually to just remove this change. `max_indices` is never going to contain a padding index--if it did, then the forward mode wouldn't be doing it's job properly. Don't know how I overlooked that lol